lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1167,15 +1167,30 @@ Value *SPIRVToLLVM::mapValue(SPIRVValue *BV, Value *V) {
   if (Loc != ValueMap.end()) {
     if (Loc->second == V)
       return V;
+
+    // Existing LLVM mapped value could be the local SPIRV Builtin variable
+    // created during forward translation. In this case, we are trying to
+    // replace with mapped value with the global SPIRV Builtin variable
+    // (passed via V here). In this case, we also need a AddrSpaceCast to cast
+    // between the new mapped value and the old.
+    if (Loc->second->hasName() &&
+        Loc->second->getName().starts_with("__local_SPIRV_Builtin")) {
```

Comment:
If reader counterpart is not done, are regressions expected?

Reply:
I don't think so...I can try it out in a separate PR.